comit_node/src/swap_protocols/actions.rs

```diff
@@ -52,12 +55,57 @@ pub mod ethereum {
         pub network: Network,
     }
 
-    #[derive(Debug, Clone, PartialEq, Serialize)]
+    #[derive(Debug, Clone, PartialEq)]
     pub struct CallContract {
         pub to: Address,
-        pub data: Bytes,
+        pub data: Option<Bytes>,
         pub gas_limit: U256,
         pub network: Network,
         pub min_block_timestamp: Option<Timestamp>,
     }
+
+    impl Serialize for CallContract {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            let mut state = serializer.serialize_struct("CallContract", 5)?;
+
+            state.serialize_field("to", &self.to)?;
+            if let Some(data) = &self.data {
+                state.serialize_field("data", data)?;
+            }
+            state.serialize_field("gas_limit", &self.gas_limit)?;
+            state.serialize_field("network", &self.network)?;
+            if let Some(min_block_timestamp) = &self.min_block_timestamp {
+                state.serialize_field("min_block_timestamp", min_block_timestamp)?;
+            }
+
+            state.end()
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use ethereum_support::{web3::types::U256, Address, Network};
+    use std::str::FromStr;
+
+    #[test]
+    fn call_contract_serializes_correctly_to_json_with_none() {
```

Comment:
This test should be testing what is now actually serialized to JSON, which is `ActionResponsePayload::EthereumCallContract`.